src/App.php

```diff
@@ -1033,7 +1034,7 @@ public function setPConfigValue($uid, $cat, $k, $v)
 	{
 		// Only arrays are serialized in database, so we have to unserialize sparingly
 		$value = is_string($v) && preg_match("|^a:[0-9]+:{.*}$|s", $v) ? unserialize($v) : $v;
-
+		$this->setConfigRecursively([$uid, $cat]);
```

Comment:
This looks way too arcane, and sometimes simpler is better. I'd much rather see something like:
```php
if (!isset($this->config[$uid])) {
    $this->config[$uid] = [];
}

if (!isset($this->config[$uid][$cat])) {
    $this->config[$uid][$cat] = [];
}
```

Instead of this function call.